src/webgpu/api/validation/encoding/render_bundle.spec.ts

```diff
@@ -51,3 +60,245 @@ g.test('render_bundles,device_mismatch')
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        _compatible: true,
+      }, // control case
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['bgra8unorm', 'bgra8unorm'] as const,
+        _compatible: false,
+      },
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['rg8unorm', 'bgra8unorm'] as const,
+        _compatible: false,
+      },
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm', 'rgba8unorm'] as const,
+        passFormats: ['rg8unorm', 'bgra8unorm'] as const,
+        _compatible: false,
+      },
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['rg8unorm', 'bgra8unorm', 'rgba8unorm'] as const,
+        _compatible: false,
+      },
+    ])
+  )
+  .fn(async t => {
+    const { bundleFormats, passFormats, _compatible } = t.params;
+
+    const bundleEncoder = t.device.createRenderBundleEncoder({
+      colorFormats: bundleFormats,
+    });
+    const bundle = bundleEncoder.finish();
+
+    const encoder = t.createEncoder('render pass', {
+      attachmentInfo: {
+        colorFormats: passFormats,
+      },
+    });
+    encoder.encoder.executeBundles([bundle]);
+
+    encoder.validateFinish(_compatible);
+  });
+
+g.test('depth_stencil_formats_mismatch')
+  .desc(
+    `
+    Tests executeBundles cannot be called with render bundles that do match the depthStencil of the
+    render pass. This includes:
+    - formats don't match
+    - formats have matching depth or stencil aspects, but other aspects are missing
+    `
+  )
+  .params(u =>
+    u.combineWithParams([
+      { bundleFormat: 'depth24plus' as const, passFormat: 'depth24plus' as const }, // control case
+      { bundleFormat: 'depth24plus' as const, passFormat: 'depth16unorm' as const },
+      { bundleFormat: 'depth24plus' as const, passFormat: 'depth24plus-stencil8' as const },
+      { bundleFormat: 'stencil8' as const, passFormat: 'depth24plus-stencil8' as const },
+    ])
+  )
+  .fn(async t => {
+    const { bundleFormat, passFormat } = t.params;
+
+    const compatible = bundleFormat === passFormat;
+
+    const bundleEncoder = t.device.createRenderBundleEncoder({
+      colorFormats: [],
+      depthStencilFormat: bundleFormat,
+    });
+    const bundle = bundleEncoder.finish();
+
+    const encoder = t.createEncoder('render pass', {
+      attachmentInfo: {
+        colorFormats: [],
+        depthStencilFormat: passFormat,
+      },
+    });
+    encoder.encoder.executeBundles([bundle]);
+
+    encoder.validateFinish(compatible);
+  });
+
+g.test('depth_stencil_readonly_mismatch')
+  .desc(
+    `
+    Tests executeBundles cannot be called with render bundles that do match the depthStencil
+    readonly state of the render pass.
+    `
+  )
+  .params(u =>
+    u.combine('depthStencilFormat', kDepthStencilFormats).combineWithParams([
```

Comment:
nit: newline between combine and combineWithParams

unless the autoformatter did this

Reply:
Pretty sure this was the work of the autoformatter. I wouldn't have formatted it that way.